sdk/src/beta9/runner/taskqueue.py

```diff
@@ -304,14 +304,20 @@ def process_tasks(self, channel: Channel) -> None:
                     result = None
                     duration = None
 
+                    encountered_exception = ""
+
                     try:
                         args = task.args or []
                         kwargs = task.kwargs or {}
 
                         result = handler(context, *args, **kwargs)
-                    except BaseException:
-                        print(traceback.format_exc())
-                        task_status = TaskStatus.Error
+                    except BaseException as e:
+                        if type(e) in handler.handler.parent.retry_for:
```

Comment:
it would be nice to improve the handler class so we could reference this in a way that's not so nested